model/task/task.go

```diff
@@ -3668,6 +3674,24 @@ func getTasksByVersionPipeline(versionID string, opts GetTasksByVersionOptions)
 	return pipeline
 }
 
+func addTaskDuration() bson.M {
+	// Time taken for a task is in nanoseconds, but subtracting two dates yields milliseconds.
+	convertToNanoseconds := time.Millisecond
+	return bson.M{
+		"$addFields": bson.M{
+			TaskDurationKey: bson.M{
+				"$cond": bson.M{
```

Comment:
I think it wouldn't be the worst idea to introduce the idea of a "loop through tasks to set things" for this function, especially as we continue to add things. One for loop may be less expensive than adding aggregation steps (depending on what's needed to be done).

Reply:
I also wonder if we start using TimeTaken instead, just on the UI we could check "if TimeTaken exists use that, else use Now - Start" since we'd be checking fields anyway but I'm not sure if there's some issue with that. On the legacy UI I know we have the timer that increments every second which was pretty cool.